### PR TITLE
Change the “exchange with Uniswap” button to 1inch #2265

### LINKED ISSUE #2265: Change the “exchange with Uniswap” button to 1inch
ge the “exchange with Uniswap” button to "Exchange with 1inch" 

https://1inch.exchange/#/bat/eth/#/r/0x98f21584006c79871F176F8D474958a69e04595B/

0x98f21584006c79871F176F8D474958a69e04595B is alphawallet.eth

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #2265

### CODE REVIEW

File: AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private static let refferal = "r/0x98f21584006c79871F176F8D474958a69e04595B"
+
+    private let input: Input
+
+    init(input: Input) {
+        self.input = input
+    }
+
+    var url: URL? {
+        var value = Oneinch.baseURL + Oneinch.separator + input.subpath + Oneinch.separator + Oneinch.refferal
```
Comment (hboon) [inline AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift:52]: Better to use `URLComponents`?

**Review by hboon — COMMENTED** [review]
The code doesn't seem to include the token pair symbols in the URL yet, but I understand it's WIP.

File: AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+//
+//  Created by Vladyslav Shepitko on 11.11.2020.
+//
+
+import Foundation
+import PromiseKit
+import Moya
+
+class Oneinch {
+    private(set) static var availableTokens: [Oneinch.ERC20Token] = []
```
Comment (hboon) [inline AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift:13]: Can we make this be instance-side?

**Review by hboon — COMMENTED** [review]
Is this ready for review?

But some observations:

- [ ] Can we make the button title be "Swap" instead of "Exchange with 1inch"?
- [ ] Can we get Swap button to show up for Ether too?

File: AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift (hunk 2 above)
Comment (vladyslav-iosdev) [reply AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift:13]: @hboon not clearly understand about this, can u provide more details?

Comment (hboon) [reply AlphaWallet/Tokens/Types/Oneinch/Oneinch.swift:13]: Remove `static` :)

```
private(set) var availableTokens: [Oneinch.ERC20Token] = []
```

**Review by James-Sangalli — APPROVED** [review]
@hboon this seems to work as expected, the swap button is on ETH

**Review by hboon — APPROVED** [review]
Thanks @vladyslav-iosdev @James-Sangalli !

**Pull request merged**